### PR TITLE
[FEATURE] - upgrade to latest version, Kong and KIC

### PR DESCRIPTION
**Pull request by nutellinoit**
In this PR i'm upgrading kong OSS kong-gateway EE and kong ingress controller to the latest available version + some minor upgrades to READMEs and examples. The new version will be -> `2.8.1` to follow Kong versioning

Tasks:

- [x] Upgrade KIC to 2.3.x
- [x] Upgrade Kong to 2.8.x
- [x] Align petstore examples
- [x] Align README with new standard
- [x] Check CI/CD tests

### CODE REVIEW

**Review by nohant — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,36 +1,62 @@
...
+![Slack](https://img.shields.io/badge/slack-@kubernetes/fury-yellow.svg?logo=slack&label=Slack)
 
-## Ingress Packages
+<!-- <KFD-DOCS> -->
 
-Following packages are included in Fury Kubernetes Kong katalog:
-
-- [kong](katalog/kong): The KONG Ingress Controller for Kubernetes
-provides delivery API Gateway for Kubernetes applications. Version: **1.3.1**
+*Kubernetes Fury Kong* contains Kong Ingress Controller for Kubernetes and all the possible deployment options as an
```
Comment (ralgozino) [inline README.md]:
```suggestion
**Kubernetes Fury Kong** contains Kong Ingress Controller for Kubernetes and all the possible deployment options as an
```

File: README.md [hunk 2]
```diff
@@ -1,36 +1,62 @@
...
 
 - [kong-enterprise](katalog/kong/kong-enterprise): The KONG Ingress Controller for Kubernetes with Kong Enterprise Edition.
-  Version: Ingress: **1.3.1**, Kong Enterprise: **2.4.1.1**
+  Version: Ingress: **2.3.1**, Kong Enterprise: **2.8.1.0**
 - [kong-enterprise-hybrid](katalog/kong/kong-enterprise-hybrid): The KONG Ingress Controller for Kubernetes with Kong Enterprise Edition. Data Planes connected to Control Plane in Hybrid Mode.
-  Version: Ingress: **1.3.1**, Kong Enterprise: **2.4.1.1**
+  Version: Ingress: **2.3.1**, Kong Enterprise: **2.8.1.0**
 - [kong-enterprise-k8s](katalog/kong/kong-enterprise-k8s): The KONG Ingress Controller for Kubernetes with Kong Enterprise Edition for K8s.
-  Version: Ingress: **1.3.1**, Kong Enterprise: **2.4.1.1**
+  Version: Ingress: **2.3.1**, Kong Enterprise: **2.8.1.0**
```
Comment (ralgozino) [inline README.md]: can we make this a table like we do for the rest of the modules? 
We could have 2 tables, 1 for OSS and 1 for enterprise, or we could add a column saying the package is opensource or enterprise

File: docs/releases/v2.8.1.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
+# Kong Add-on Module Release 2.8.1
+
+Welcome to the latest release of `kong` module of [`Kubernetes Fury
```
Comment (ralgozino) [inline docs/releases/v2.8.1.md]:
```suggestion
Welcome to the latest release of the `kong` add-on module for [`Kubernetes Fury
```

File: docs/releases/v2.8.1.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+Distribution`](https://github.com/sighupio/fury-distribution) maintained by team
+SIGHUP.
+
+This is a minor release with the goal to update kong images and kong ingress controllers to the latest version available.
```
Comment (ralgozino) [inline docs/releases/v2.8.1.md]: is `kong` all in underscores the right spelling or should it be `Kong`?

File: docs/releases/v2.8.1.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+| Component                            | Supported Version                                                                                      | Previous Version |
+|--------------------------------------|--------------------------------------------------------------------------------------------------------|------------------|
+| `kong`                               | [`v2.8.1`](https://github.com/Kong/kong/releases/tag/2.8.1)                                            | `v2.3.2`         |
+| `kong/kong-gateway`                  | [`v2.8.1.0`](https://docs.konghq.com/gateway/changelog/#2810)                                          | `v2.4.1.1`       |
+| `kong/kubernetes-ingress-controller` | [`v2.3.1`](https://github.com/Kong/kubernetes-ingress-controller/releases/tag/v2.3.1)                  | `v1.3.1`         |
+
+## Update Guide for db-less deployment options 🦮
+
+To update your kong cluster to the latest version, you simply need to apply the updated manifests.
+There will be some downtime while the new version is deployed.
```
Comment (ralgozino) [inline docs/releases/v2.8.1.md:20]: it doesn't do a rolling upgrade?

File: katalog/kong/README.md [hunk 6]
```diff
@@ -17,16 +17,16 @@ between control plane and data plane and adds the secret generation for the TLS
 
 ## Requirements
 
-- Kubernetes >= `1.18.0`
-- Kustomize >= `v3`
+- Kubernetes - See module compatibility matrix
```
Comment (ralgozino) [inline katalog/kong/README.md]:
```suggestion
- Kubernetes - See module [compatibility matrix](https://github.com/sighupio/fury-kubernetes-kong/blob/master/docs/COMPATIBILITY_MATRIX.md)
```

**Review by ralgozino — CHANGES_REQUESTED** [review]
Hi @nutellinoit 

I left some minor comments for the text in the docs, additionally, here are some questions that come to my mind:

- Do we need the `katalog/kong` folder at all? Being that the module is called `Kong` can we just put all the packages at the `katalog`?
- Why not put `katalog/kong/upstream` folder inside `base`?
- Should add a `MAINTENANCE.md` file?

Thanks!

File: docs/releases/v2.8.1.md (hunk 5 above)
Comment (nutellinoit) [reply docs/releases/v2.8.1.md:20]: Yes, but our "deploy" is a DaemonSet with attached services with `externalTrafficPolicy: Local` and in some cases you can have little downtimes

File: docs/releases/v2.8.1.md (hunk 4 above)
Comment (nutellinoit) [reply docs/releases/v2.8.1.md]: it's Kong, with capital K

File: README.md (hunk 2 above)
Comment (nutellinoit) [reply README.md]: Yes, we can

**Comment by nutellinoit** [comment]
Hi @ralgozino !

> * Do we need the `katalog/kong` folder at all? Being that the module is called `Kong` can we just put all the packages at the `katalog`?

I prefer to do that on a next major version, Kong 3.0.0 is coming soon

> * Why not put `katalog/kong/upstream` folder inside `base`?

Yes, we can

> * Should add a `MAINTENANCE.md` file?

Yes, but in a different PR, i will open an issue to track it

**Comment by ralgozino** [comment]
sounds good to me

File: README.md (hunk 2 above)
Comment (nutellinoit) [reply README.md]: Done ✅

File: docs/releases/v2.8.1.md (hunk 4 above)
Comment (nutellinoit) [reply docs/releases/v2.8.1.md]: Done ✅

**Review by ralgozino — APPROVED** [review]
lgtm

**Pull request merged**